app/controllers/decidim/participatory_processes/participatory_processes_controller.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
```

Comment:
Instead of copying over the whole file, we can create a concern as described in `docs/MODIFYING_RUBY_ENGINE.rb`. (You can also check out the [official docs](https://api.rubyonrails.org/classes/ActiveSupport/Concern.html) here.) 

Add a new file at `app/controllers/concerns/participatory_processes_controller_extensions.rb` (the path is important; the filename doesn't matter, but naming it like this this helps us stay organized). In that file, we only need to include what we want to override:

```ruby
# frozen_string_literal: true

require "active_support/concern"

module ParticipatoryProcessesControllerExtensions
  include Decidim::ParticipatoryProcesses
  extend ActiveSupport::Concern

  included do
    def index
     raise ActionController::RoutingError, "Not Found" if published_processes.none?

     enforce_permission_to :list, :process
     enforce_permission_to :list, :process_group

    # redirect to the process if there's only one published process
     if published_processes.count == 1
        redirect_to action: "show", slug:published_processes.first.slug
      end
    end

    private
    
    def published_processes
      @published_processes ||= Decidim::ParticipatoryProcesses::OrganizationPublishedParticipatoryProcesses.new(current_organization, current_user)
    end
  end
end
```
 Then, in `application.rb`, add:

`Decidim::ParticipatoryProcesses::ParticipatoryProcessesController.send(:include, ParticipatoryProcessesControllerExtensions)`